src/OSPSuite.Core/Snapshots/Mappers/ProjectMapper.cs

```diff
@@ -65,21 +73,36 @@ protected Task<TSnapshot[]> MapModelsToSnapshot<TModel, TSnapshot>(IEnumerable<T
          return AwaitAs<TSnapshot>(tasks);
       }
 
-      protected Task<SnapshotDataRepository[]> MapObservedDataToSnapshots(IReadOnlyCollection<ModelDataRepository> allObservedData) => 
+      protected virtual async Task<ParameterIdentification[]> MapParameterIdentificationToSnapshots(IReadOnlyCollection<ModelParameterIdentification> allParameterIdentifications)
+      {
+         return await _parameterIdentificationMapper.MapToSnapshots(allParameterIdentifications);
```

Comment:
return await s always suspicious unless there is a try catch. Just returm the task